ui/tsconfig.json

```diff
@@ -22,6 +22,12 @@
     "module": "es2015",
     "baseUrl": "./",
     "jsx": "react",
-    "resolveJsonModule": true
+    "resolveJsonModule": true,
+    "skipLibCheck": true
+    // "noUnusedLocals": false,
+  },
+  "angularCompilerOptions": {
+    "disableTypeScriptVersionCheck": true,
```

Comment:
Allows compile to complete despite warning from @angular/compiler-cli.
<img width="688" alt="Screen Shot 2021-03-17 at 1 36 15 PM" src="https://user-images.githubusercontent.com/35533885/111512151-d3b00500-8725-11eb-8160-f10c5225e533.png">

Reply:
mm, this warning is very concerning - I'm not sure it's worth moving forward with this without pushing the Angular upgrade through. Based on this version mapping I found - to get to typescript 4.3, you will need to go to Angular 12: https://gist.github.com/LayZeeDK/c822cc812f75bb07b7c55d07ba2719b3

If the Angular upgrade is easy (i.e. < .5 day), then it may be worth pursuing. Otherwise - we can pause this PR until Angular removal (in progress again), and perform this upgrade after.

Even if it appears to work locally, the warning indicates that there may be some optimizations that the Angular compiler is making (most of these things are not enabled locally) which could diverge from what the Typescript compiler is doing. I think it could be very difficult to debug this kind of issue, and the only fix at that point would probably be to revert this PR, then upgrade Angular when fixing forward. I am especially paranoid of issues like this since it's possible they would avoid detection until runtime, and they are difficult to automatically report/track.

Reply:
I would like to upgrade Angular too because many libs are interdependent. I found instr for how to upgrade from Angular 7.2 to the latest. What do you think?
https://update.angular.io/?l=3&v=7.2-12.0

Reply:
Good find. Long list of things here, though many will likely not apply. Scanning the list, my biggest concern is the removal of `Http`. Our Angular swagger codegen currently depends on this, and I don't know whether upgrading to a newer codegen version will fix it (alternately, upgrading our swagger codegen version will likely cause other problems). However, we are actually in a decent position to kill all the Angular codegen right now - I think there's a path through the current remaining code to do this. @als364 FYI

Reply:
Thanks for the heads up. I still expect that at the current rate of conversion we would be 1-2 months before we are actually in a position to delete all the angular codegen.

Reply:
There aren't many references remaining to the Angular swagger codegen, the 1-2 month presumably assumes we convert all services before removing it. There is an alternate path here where simply convert the remaining references in-place - this may actually help to reduce the scope of the ongoing service conversions. There's 25 files still referencing these by my count, and most are test files (many siloed and non-central). Looks like a couple of hours of work to me unless there are complications (a lot of this will just be search/replace, since the models are effectively interchangeable).

Reply:
Ooh, yeah, you're right about what I was thinking. If you're thinking just switch everything to `fetch` I can take a crack at that on Monday.

Reply:
Could we push this PR through with the code changes, but not the actual typescript upgrade? Or will that cause issues with our current version of TS? If we can keep the code changes, we can keep the work that went into the conversion, making it a smaller PR when we are ready to actually up the version.

Reply:
Sounds good to me.